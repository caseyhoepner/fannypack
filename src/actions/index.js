export const setWeather = (weatherData) => ({
  type: 'SET_WEATHER', 
  weatherData
})