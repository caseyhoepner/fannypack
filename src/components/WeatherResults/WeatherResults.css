.wr-icon {
  width: 100px;
}

.wr-container {
  display: flex;
  width: 80%;
  margin: auto;
  justify-content: space-between;
}

.wr-weather-container {

}

.wr-pack-container {

}