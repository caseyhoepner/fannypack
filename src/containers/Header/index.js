import React, { Component } from 'react';
import { connect } from 'react-redux';

import './Header.css'

export class Header extends Component {
  render() {
    return (
      <div className='h-header-container'>
        <h1 className='h-title'>FannyPack</h1>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
})

export default connect(null, mapDispatchToProps)(Header);