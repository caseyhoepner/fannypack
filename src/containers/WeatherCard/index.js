import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { packingData } from '../../utils/packingData';
import './WeatherCard.css';

export const WeatherCard = ({ highTemp, lowTemp, summary, icon, city, day}) => {

  const getEssentials = () => {
    let essentials = [];

    if (highTemp >= 80) {
      essentials = [...essentials, ...packingData.hot]
    }

    if (highTemp < 80 && lowTemp > 70) {
      essentials = [...essentials, ...packingData.warm]
    }

    if (highTemp < 70 && lowTemp > 40) {
      essentials = [...essentials, ...packingData.cool]
    }

    if (lowTemp <= 40) {
      essentials = [...essentials, ...packingData.cold]
    }

    essentials = [...essentials, ...packingData[icon]];

    const filteredEssentials = filterEssentials(essentials)

    return filteredEssentials.map(filteredEssential => {
      return <li className='wc-essentials-list'key={filteredEssential}>{filteredEssential}</li>
      })
  }

  const filterEssentials = (essentials) => {
    return essentials.reduce((filteredEssentials, essential) => {
      if (!filteredEssentials.includes(essential)) {
        filteredEssentials.push(essential);
      }
      return filteredEssentials;
    }, []);
  }

  const formatCity = function (city) {
    city = city.toLowerCase().split(' ');
    for (var i = 0; i < city.length; i++) {
      city[i] = city[i].charAt(0).toUpperCase() + city[i].slice(1);
    }
    return city.join(' ');
  };

  return (
    <div className="wc-container">
      <div className='wc-weather-container'>
        <h2 className='wc-day'>{`Day ${day}: ${formatCity(city)}`}</h2>
          <section className='wc-temp-icon-container'>
            <img className='wc-icon' src={require(`../../assets/${icon}.svg`)} />
            <section className='wc-temp-container'>
              <p className='wc-high'>High: {highTemp.toFixed(0)}&#176;</p>
              <p className='wc-low'>Low: {lowTemp.toFixed(0)}&#176;</p>
            </section>
          </section>  
          <p className='wc-summary'>{summary}</p>
      </div>
      <section className='wc-essentials'>
        <h3 className='wc-essentials-title'>What to Pack:</h3>
          <ul className='wc-essentials-list'>
            { getEssentials() }
          </ul>
      </section>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  weatherData: state.weatherData
})

WeatherCard.propTypes = {
  weatherData: PropTypes.array
};

export default withRouter(connect(mapStateToProps)(WeatherCard));