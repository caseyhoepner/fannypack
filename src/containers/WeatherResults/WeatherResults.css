.wr-container {
  display: grid;
  grid-gap: 15px;
  grid-template-columns: repeat(auto-fit, minmax(260px, 1fr));
  margin: 20px auto;
  width: 70%;
}

.wr-loading-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 70px;
}

.wr-suitcase {
  height: 100px;
  margin: auto;
}

.typewriter h1 {
  color: #fff;
  padding: 20px 0 0 0;
  overflow: hidden;
  border-right: .15em solid orange;
  white-space: nowrap;
  margin: 20px auto;
  letter-spacing: .2em;
  animation: 
    typing 3.5s steps(30, end),
    blinking-cursor .5s step-end infinite;
}

@keyframes typing {
  from { width: 0 }
  to { width: 100% }
}

@keyframes blinking-cursor {
  from, to { border-color: transparent }
  50% { border-color: orange; }
}