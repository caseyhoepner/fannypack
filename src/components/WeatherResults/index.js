import React from 'react';
import { connect } from 'react-redux';
import { Header } from '../../containers/Header'

import './WeatherResults.css'

const WeatherResults = (props) => {
  const { highTemp, lowTemp, icon, summary } = props.weatherData;
  return (
    <div>
      <Header />
        <div className='wr-container'>
          <div className='wr-weather-container'>
            <h2>Weather</h2>
              <p>High: {highTemp}</p>
              <p>Low: {lowTemp}</p>
              <p>{summary}</p>
              <img className='wr-icon' src={require(`../../assets/rain.svg`)} />
          </div>
          <div className='wr-pack-container'>
            <h2>What to Pack</h2>
          </div>
        </div>
    </div>
  )
}

const mapStateToProps = (state) => ({
  weatherData: state.weatherData
})

export default connect(mapStateToProps)(WeatherResults);