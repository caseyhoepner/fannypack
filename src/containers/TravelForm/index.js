import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { setWeather, changeToLoaded } from '../../actions';
import Header from '../../containers/Header';
import * as API from '../../utils/';
import './TravelForm.css';

export class TravelForm extends Component {
  constructor() {
    super() 
    this.state = {
      day1: '',
      day2: '',
      day3: '',
      day4: '',
      day5: '',
      day6: '',
      day7: ''
    }
  }

  handleChange = (event) => {
    event.preventDefault();
    const { name, value } = event.target;
    this.setState({[name]: value});
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.history.push('/results');
    this.getWeatherData();
  }

  getWeatherData = async () => {
    const { day1, day2, day3, day4, day5, day6, day7 } = this.state;
    const { setWeather, changeToLoaded } = this.props;

    // for(let i = 1; i < 8; i++) {
    //   console.log(`day${i}`)
    //   if (`day${i}`) {
    //     const weatherData = await API.fetchWeather(`day${i}, ${i}`);
    //     setWeather(weatherData, `day${i}, ${i}`) 
    //   }
    // }

    if (day1) {
      const weatherData = await API.fetchWeather(day1, 1);
      setWeather(weatherData, day1, 1);
    }

    if (day2) {
      const weatherData = await API.fetchWeather(day2, 2);
      setWeather(weatherData, day2, 2);
    }

    if (day3) {
      const weatherData = await API.fetchWeather(day3, 3);
      setWeather(weatherData, day3, 3);
    }

    if (day4) {
      const weatherData = await API.fetchWeather(day4, 4);
      setWeather(weatherData, day4, 4);
    }

    if (day5) {
      const weatherData = await API.fetchWeather(day5, 5);
      setWeather(weatherData, day5, 5);
    }

    if (day6) {
      const weatherData = await API.fetchWeather(day6, 6);
      setWeather(weatherData, day6, 6);
    }

    if (day7) {
      const weatherData = await API.fetchWeather(day7, 7);
      setWeather(weatherData, day7, 7);
    }
    
    changeToLoaded(true);
  }

  render() {
    const { day1, day2, day3, day4, day5, day6, day7 } = this.state;

    return (
      <div className='parent-container'>
        <Header />
        <form className='tf-form'onSubmit={this.handleSubmit}>
          <h2 className='tf-title'>Where are you going?</h2>
            <div className='tf-form-components'>
              <section className='tf-tomorrow tf-day-input'>
                <p className='tf-day'>Tomorrow</p>
                  <input 
                    className='test-input tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day1'
                    value={day1}
                  />
              </section>
              <section 
                className='tf-day-2 tf-day-input'>
                <p className='tf-day'>Day 2</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day2'
                    value={day2}
                  />
              </section>              
              <section 
                className='tf-day-3 tf-day-input'>
                <p className='tf-day'>Day 3</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day3'
                    value={day3}
                  />
              </section>              
              <section className='tf-day-4 tf-day-input'>
                <p className='tf-day'>Day 4</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day4'
                    value={day4}
                  />
              </section>              
              <section className='tf-day-5 tf-day-input'>
                <p className='tf-day'>Day 5</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day5'
                    value={day5}
                  />
              </section>              
              <section className='tf-day-6 tf-day-input'>
                <p className='tf-day'>Day 6</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day6'
                    value={day6}
                  />
              </section>              
              <section className='tf-day-7 tf-day-input'>
                <p className='tf-day'>Day 7</p>
                  <input 
                    className='tf-form-component tf-input-component'
                    onChange={this.handleChange}
                    placeholder='Destination'
                    name='day7'
                    value={day7}
                  />
              </section>              
              <button 
                className='tf-form-component tf-btn'>
                Get Packing List
              </button>
            </div>
          </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setWeather: (weatherData, city, day) => dispatch(setWeather(weatherData, city, day)), 
  changeToLoaded: (bool) => dispatch(changeToLoaded(bool))
});

TravelForm.propTypes = {
  setWeather: PropTypes.func.isRequired,
  changeToLoaded: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(TravelForm);