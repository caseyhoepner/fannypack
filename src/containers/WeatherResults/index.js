import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Header from '../../components/Header';
import WeatherCard from '../../containers/WeatherCard';
import suitcase from '../../assets/suitcase.svg';
import './WeatherResults.css';
import  uuid from 'uuid';

export const WeatherResults = ({ weatherData, isLoaded }) => {
  let results;

  if (!isLoaded) {
    results = 
      <section className='wr-loading-container'>
        <img className='wr-suitcase' src={suitcase} alt='Icon of a suitcase.'/>
        <div className='typewriter'>
          <h1>Compiling your packing lists...</h1>
        </div>
      </section>
  } else {
    results = weatherData.map(city => {
      return <WeatherCard {...city} key={uuid()}/>;
    })
  };

    return (
      <div>
        <Header />
          <div className='wr-container'>
            { results }
          </div>
      </div>
    )
};

export const mapStateToProps = (state) => ({
  weatherData: state.weatherData,
  isLoaded: state.isLoaded
});

WeatherResults.propTypes = {
  weatherData: PropTypes.array.isRequired,
  isLoaded: PropTypes.bool.isRequired,
};

export default withRouter(connect(mapStateToProps)(WeatherResults));