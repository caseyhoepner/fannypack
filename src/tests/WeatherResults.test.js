import React from 'react'
import { shallow, mount } from 'enzyme'
import { mockWeatherData } from './testMocks';
import { WeatherResults, mapStateToProps } from '../containers/WeatherResults'
import * as Actions from '../actions'
jest.mock('uuid', () => jest.fn(() => 1));

describe('WeatherResults', () => {
  describe('WeatherResults Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<WeatherResults weatherData={[mockWeatherData]} isLoaded={true}/>);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot', () => {
      let wrapper = shallow(<WeatherResults weatherData={[mockWeatherData]} isLoaded={false}/>);
      
      expect(wrapper).toMatchSnapshot()
    })

  describe('mapStateToProps', () => {
    it('should parse the weatherData from state', () => {
      const mockState = {
        weatherData: mockWeatherData
      }

      const expected = mockWeatherData

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.weatherData).toEqual(expected)
    })

    it('should parse isLoaded from state', () => {
      const mockState = {
        isLoaded: true
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.isLoaded).toEqual(true)
    })
  })
})
})