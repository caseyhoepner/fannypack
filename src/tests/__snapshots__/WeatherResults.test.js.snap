// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherResults WeatherResults Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherResults
    isLoaded={true}
    weatherData={
      Array [
        Object {
          "city": "Denver",
          "day": 1,
          "highTemp": 70,
          "icon": "rain-clouds.svg",
          "lowTemp": 50,
          "summary": "Mostly cloudy.",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="wr-container"
        >
          <withRouter(Connect(WeatherCard))
            city="Denver"
            day={1}
            highTemp={70}
            icon="rain-clouds.svg"
            lowTemp={50}
            summary="Mostly cloudy."
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Connect(WeatherCard))
              city="Denver"
              day={1}
              highTemp={70}
              icon="rain-clouds.svg"
              lowTemp={50}
              summary="Mostly cloudy."
            />,
          ],
          "className": "wr-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "city": "Denver",
              "day": 1,
              "highTemp": 70,
              "icon": "rain-clouds.svg",
              "lowTemp": 50,
              "summary": "Mostly cloudy.",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="wr-container"
          >
            <withRouter(Connect(WeatherCard))
              city="Denver"
              day={1}
              highTemp={70}
              icon="rain-clouds.svg"
              lowTemp={50}
              summary="Mostly cloudy."
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <withRouter(Connect(WeatherCard))
                city="Denver"
                day={1}
                highTemp={70}
                icon="rain-clouds.svg"
                lowTemp={50}
                summary="Mostly cloudy."
              />,
            ],
            "className": "wr-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "city": "Denver",
                "day": 1,
                "highTemp": 70,
                "icon": "rain-clouds.svg",
                "lowTemp": 50,
                "summary": "Mostly cloudy.",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`WeatherResults WeatherResults Component should match the snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherResults
    isLoaded={false}
    weatherData={
      Array [
        Object {
          "city": "Denver",
          "day": 1,
          "highTemp": 70,
          "icon": "rain-clouds.svg",
          "lowTemp": 50,
          "summary": "Mostly cloudy.",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="wr-container"
        >
          <section
            className="wr-loading-container"
          >
            <img
              className="wr-suitcase"
              src="suitcase.svg"
            />
            <div
              className="typewriter"
            >
              <h1>
                Compiling your packing lists...
              </h1>
            </div>
          </section>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="wr-loading-container"
          >
            <img
              className="wr-suitcase"
              src="suitcase.svg"
            />
            <div
              className="typewriter"
            >
              <h1>
                Compiling your packing lists...
              </h1>
            </div>
          </section>,
          "className": "wr-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="wr-suitcase"
                src="suitcase.svg"
              />,
              <div
                className="typewriter"
              >
                <h1>
                  Compiling your packing lists...
                </h1>
              </div>,
            ],
            "className": "wr-loading-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "wr-suitcase",
                "src": "suitcase.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Compiling your packing lists...
                </h1>,
                "className": "typewriter",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Compiling your packing lists...",
                },
                "ref": null,
                "rendered": "Compiling your packing lists...",
                "type": "h1",
              },
              "type": "div",
            },
          ],
          "type": "section",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="wr-container"
          >
            <section
              className="wr-loading-container"
            >
              <img
                className="wr-suitcase"
                src="suitcase.svg"
              />
              <div
                className="typewriter"
              >
                <h1>
                  Compiling your packing lists...
                </h1>
              </div>
            </section>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="wr-loading-container"
            >
              <img
                className="wr-suitcase"
                src="suitcase.svg"
              />
              <div
                className="typewriter"
              >
                <h1>
                  Compiling your packing lists...
                </h1>
              </div>
            </section>,
            "className": "wr-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  className="wr-suitcase"
                  src="suitcase.svg"
                />,
                <div
                  className="typewriter"
                >
                  <h1>
                    Compiling your packing lists...
                  </h1>
                </div>,
              ],
              "className": "wr-loading-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "wr-suitcase",
                  "src": "suitcase.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    Compiling your packing lists...
                  </h1>,
                  "className": "typewriter",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Compiling your packing lists...",
                  },
                  "ref": null,
                  "rendered": "Compiling your packing lists...",
                  "type": "h1",
                },
                "type": "div",
              },
            ],
            "type": "section",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
